### PR TITLE
Tech/migrating micro deps to jdk7

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
@szpak @nurkiewicz @mchmielarz @dst or anyone else can you please review ASAP so that I don't have gigantic problems with merging? :P

### CODE REVIEW

**Comment by dst** [comment]
630 additions and 354 deletions. 
The statistics are so poor that I cannot look inside the code.

**Comment by marcingrzejszczak** [comment]
Welcome to the Java world :)

File: micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/ServicePath.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        this.path = path;
+    }
+
+    public String getPath() {
+        return path;
+    }
+
+    @Override
+    public String toString() {
+        return ReflectionToStringBuilder.toString(this);
```
Comment (nurkiewicz) [inline micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/ServicePath.java]: Do we really need all that reflection here? It's just one field, why not be explicit what we show/compare?

**Comment by marcingrzejszczak** [comment]
@dst @nurkiewicz @szpak - I'll merge this code tomorrow evening (don't want to diverge like the last time when I migrated code to Java).

**Comment by szpak** [comment]
@marcingrzejszczak That's your project. Do whatever you like...

**Comment by marcingrzejszczak** [comment]
@szpak it's not my project, it's our project. But the last time I've done a big change and it was completely ignored thus this time I'm just proposing some deadline.

**Comment by marcingrzejszczak** [comment]
Anything, anyone?

**Pull request merged**